src/main/java/org/networkcalculus/dnc/AnalysisConfig.java

```diff
@@ -36,188 +37,212 @@
+	}
+
+	public void setArrivalBoundMethod(ArrivalBoundMethod arrival_bound_method) {
+		clearArrivalBoundMethods();
+		arrival_bound_methods.add(arrival_bound_method);
+	}
+
+	public Set<ArrivalBoundMethod> arrivalBoundMethods() {
+		return new HashSet<ArrivalBoundMethod>(arrival_bound_methods);
+	}
+
+	public void setArrivalBoundMethods(Set<ArrivalBoundMethod> arrival_bound_methods_set) {
+		clearArrivalBoundMethods();
+		arrival_bound_methods.addAll(arrival_bound_methods_set);
+	}
+
+	public void addArrivalBoundMethod(ArrivalBoundMethod arrival_bound_method) {
+		arrival_bound_methods.add(arrival_bound_method);
+	}
+
+	public void addArrivalBoundMethods(Set<ArrivalBoundMethod> arrival_bound_methods_set) {
+		arrival_bound_methods.addAll(arrival_bound_methods_set);
+	}
+
+	public boolean removeArrivalBoundMethod(ArrivalBoundMethod arrival_bound_method) {
+		if (arrival_bound_methods.size() == 1) {
+			return false;
+		} else {
+			return arrival_bound_methods.remove(arrival_bound_method);
+		}
+	}
+
+	public boolean convolveAlternativeArrivalBounds() {
+		return convolve_alternative_arrival_bounds;
+	}
+
+	public void setConvolveAlternativeArrivalBounds(boolean convolve_alt_abs) {
+		convolve_alternative_arrival_bounds = convolve_alt_abs;
+	}
+
+	public boolean serverBacklogArrivalBound() {
+		return server_backlog_arrival_bound;
+	}
+
+	public void setServerBacklogArrivalBound(boolean server_backlog_arrival_bound) {
+		this.server_backlog_arrival_bound = server_backlog_arrival_bound;
+	}
+
+	public boolean useFlowProlongation() {
+		return flow_prolongation;
+	}
+
+	public void setUseFlowProlongation(boolean prolong_flows) {
+		flow_prolongation = prolong_flows;
+	}
+
+	/**
+	 * Returns a deep copy of this analysis configuration.
+	 *
+	 * @return The copy.
+	 */
+	public AnalysisConfig copy() { // deep copy as primitive data types are copied by value
+		return new AnalysisConfig(multiplexing_enforcement, enforce_max_sc, enforce_max_sc_output_rate, arrival_bound_methods,
+				convolve_alternative_arrival_bounds, server_backlog_arrival_bound);
+	}
+
+	@Override
+	public int hashCode() {
+		return Objects.hash(multiplexing_enforcement, enforce_max_sc, enforce_max_sc_output_rate, arrival_bound_methods,
+				convolve_alternative_arrival_bounds, server_backlog_arrival_bound);
+	}
+
+	@Override
+	public boolean equals(Object obj) {
+		if (this == obj)
+			return true;
+		if (obj == null)
+			return false;
+		if (getClass() != obj.getClass())
+			return false;
+		AnalysisConfig other = (AnalysisConfig) obj;
+
+		return Objects.equals(this.multiplexing_enforcement, other.multiplexing_enforcement) &&
+				Objects.equals(this.enforce_max_sc, other.enforce_max_sc) &&
+				Objects.equals(this.enforce_max_sc_output_rate, other.enforce_max_sc_output_rate) &&
+				Objects.equals(this.arrival_bound_methods, other.arrival_bound_methods) &&
+				Objects.equals(this.convolve_alternative_arrival_bounds, other.convolve_alternative_arrival_bounds) &&
```

Comment:
For arrays I am doubt that Objects.equals works fine (checking all values inside). Even Javadoc recommends using deepEquals for Arrays. Could you please check? I would rather recommend using Arrays.equals here...

Reply:
Sorry, I did not find any array in this class. Please, can you specify the file and the line that you are speaking about?

Reply:
ie. Set<ArrivalBoundMethod> arrival_bound_methods- this is not an Array object but set elements need to be checked one-by-one.

Reply:
Does this coincide with my concern above?